test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -2616,13 +2666,24 @@ static void traffic_mngr_test_wred_profile(void)
 static int set_shaper(const char    *node_name,
 		      const char    *shaper_name,
 		      const uint64_t commit_bps,
-		      const uint64_t commit_burst_in_bits)
+		      const uint64_t commit_burst_in_bits,
+		      bool clamp)
 {
 	odp_tm_shaper_params_t shaper_params;
 	odp_tm_shaper_t        shaper_profile;
 	odp_tm_node_t          tm_node;
+	uint64_t               clamp_bps;
+	uint64_t               clamp_burst_in_bits;
```

Comment:
These are unnecessary and cause additional changes. Since the point is to sometimes clamp the values of the pass-by-value parameters, just assign to the parameters, i.e. this is the only thing needed:
```
        if (clamp) {
                commit_bps = clamp_rate(commit_bps);
                commit_burst_in_bits = clamp_burst(commit_burst_in_bits);
        }
```

Actually, making it conditional whether to clamp the values or not is rather dubious. If the values are already in the range (as is the case if/when this gets called by test_shaper_bw() or test_query_functions()), then the claming does not change anything, so it can as well be done. If the values are outside the range, then not clamping them means that unsupported values gets passed to ODP, violating the API spec.

So one starts to wonder why there is the clamp parameter if not clamping the values is not actually ever needed. It turns out that set_shaper() with zero rate and burst parameters are used for disabling the shaper.

The problem (and a pre-existing API spec violation in the validation test) is that set_shaper() just passes the zero values to ODP as a way to disable shaper, even if zero values are outside the valid range and even though ODP API does not specify that as a method of disabling a shaper. 

The proper way to disable a shaper is to call odp_tm_node_shaper_config() or odp_tm_queue_shaper_config() with the shaper profile parameter set to ODP_TM_INVALID.

I propose this bug to be fixed in a separate patch before the validation test changes for the API change at hand. The fix patch can be part of the same patch series in this PR. After the fix, the clamp parameter is no longer needed.

Reply:
>These are unnecessary and cause additional changes. Since the point is to sometimes clamp the values of the pass-by-value parameters, just assign to the parameters, i.e. this is the only thing needed:
Actually commit_bps and commit_burst_bps is const. 

>The proper way to disable a shaper is to call odp_tm_node_shaper_config() or odp_tm_queue_shaper_config() with the shaper profile parameter set to ODP_TM_INVALID.


Yes, the main reason was to avoid it for shaper rate zero case. 
I'll change the shaper disable logic.

Reply:
> Actually commit_bps and commit_burst_bps is const.

Indeed, I did not notice. But since they are not pointers, they can simply be changed to non-consts to avoid having to introduce new local variables as a sort-of workaround to change the values that are declared consts.